src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1287,7 +1287,7 @@ GenTree* Compiler::impBaseIntrinsic(NamedIntrinsic        intrinsic,
         {
             assert(sig->numArgs == 2);
 
-            if ((simdSize != 32) || varTypeIsFloating(simdBaseType) || compExactlyDependsOn(InstructionSet_AVX2))
+            if ((simdSize != 32) || compExactlyDependsOn(InstructionSet_AVX2))
```

Comment:
I wonder if it's worth having such ISA requirements done in a declarative way and checking that before the `switch`? (Obviously, as a follow-up exercise).

Reply:
I've considered the same. Part of the issue is the number of variations/considerations that might come into play here and things being missed in refactorings or changes.

It'd be great if we had a way to easily unit test these and validate when something is treated as intrinsic vs not. There was a functioning (but hacky) solution done for Vector<T> that was never done for the hardware intrinsics; but we're also already hitting various timing issues.

It might work better if many of the HWIntrinsic tests were ported to be library tests instead and we limited the runtime side to super simple validation of things like "was it jitted or treated as an intrinsic"...

Reply:
> It might work better if many of the HWIntrinsic tests were ported to be library tests 

What are the library tests doing differently that would make it work better?

Reply:
> What are the library tests doing differently that would make it work better?

For starters, they don't run under GCStress today. But on top of that, the managed testing frameworks already handle things like parallelizing the test runs; tracking each scenario individually when considering timeouts, etc.

xUnit (as do others like MSTest and NUnit) already handles a large number of these scenarios/considerations and so we'd be able to still provide the desired coverage while limiting what gets stress tested for the runtime to just the scenarios we believe are "most impactful".

Reply:
So, I imagine moving most of the logic to the libraries side may actually speed up the inner loop test time, make it easier to debug/validate certain changes, and fix the GCStress timeouts.

We'd still have to keep some scenarios to ensure that we do get stress test coverage; but we may be able to limit that to a single scenario per intrinsic rather than 10-12 scenarios per intrinsic.

Reply:
> For starters, they don't run under GCStress today

That is a bug, not a feature. I would love to have a capability to run GCStress on anything. One the problems with GCStress and libraries tests is reflection-based xunit runner that is too heavy for GCStress.

> managed testing frameworks already handle things like parallelizing the test runs; tracking each scenario individually when considering timeouts, etc.

We are moving the runtime tests to the same xunit authoring as libraries tests (except that the runner is going to be source generated). I hope we are going to have similar flexibility in the runtime tests once that is all done.

Reply:
> That is a bug, not a feature. I would love to have a capability to run GCStress on anything. One of the problems with GCStress and libraries tests is reflection-based xunit runner that is too heavy for GCStress.

fwiw, I created (a long time ago) GCStress pipelines for libraries tests: https://dev.azure.com/dnceng/public/_build?definitionId=839, https://dev.azure.com/dnceng/public/_build?definitionId=840, but never enabled them because there are too many timeouts and other GCStress-related issues that need to be cleaned up before they wouldn't be too noisy.